src/shared/cgroup-setup.c

```
@@ -149,51 +149,59 @@ int cg_blkio_weight_parse(const char *s, uint64_t *ret) {
-        return 0;
+        return RECURSE_DIR_CONTINUE;
 }
 
 int cg_trim(const char *controller, const char *path, bool delete_root) {
         _cleanup_free_ char *fs = NULL;
-        int r, q;
+        int r;
 
         assert(path);
 
         r = cg_get_path(controller, path, NULL, &fs);
         if (r < 0)
                 return r;
 
-        errno = 0;
-        if (nftw(fs, trim_cb, 64, FTW_DEPTH|FTW_MOUNT|FTW_PHYS) != 0) {
-                if (errno == ENOENT)
-                        r = 0;
-                else
-                        r = errno_or_else(EIO);
-        }
+        r = recurse_dir_at(
+                        AT_FDCWD,
+                        fs,
+                        /* statx_mask= */ 0,
+                        /* n_depth_max= */ UINT_MAX,
+                        RECURSE_DIR_ENSURE_TYPE,
+                        trim_cb,
+                        NULL);
+        if (r < 0)
+                log_debug_errno(r, "Failed to trim cgroup %s: %m", path);
+
+        if (delete_root && !empty_or_root(path)) {
 
-        if (delete_root) {
                 if (rmdir(fs) < 0 && errno != ENOENT)
                         return -errno;
         }
 
-        q = cg_hybrid_unified();
-        if (q < 0)
-                return q;
-        if (q > 0 && streq(controller, SYSTEMD_CGROUP_CONTROLLER)) {
-                q = cg_trim(SYSTEMD_CGROUP_CONTROLLER_LEGACY, path, delete_root);
+        if (streq_ptr(controller, SYSTEMD_CGROUP_CONTROLLER)) {
+                int q;
+
+                q = cg_hybrid_unified();
                 if (q < 0)
-                        log_warning_errno(q, "Failed to trim compat systemd cgroup %s: %m", path);
+                        return q;
+                if (q > 0) {
+                        q = cg_trim(SYSTEMD_CGROUP_CONTROLLER_LEGACY, path, delete_root);
+                        if (q < 0)
+                                log_debug_errno(q, "Failed to trim compat systemd cgroup %s: %m", path);
```

Also here. The error will be ignored.